sw/airborne/modules/ins/ins_flow.c

```diff
@@ -0,0 +1,1577 @@
+  moment = 0;
+#else
+
+  /*
+  moments[moment_ind] = Ix *(-0.000553060716181365 * (stabilization_cmd[COMMAND_ROLL]-ins_flow.lp_roll_command) -3.23315441805895 * OF_X[OF_ANGLE_DOT_IND]);
+
+  int select_ind = moment_ind - MOMENT_DELAY;
+  if(select_ind < 0) {
+  select_ind += MOMENT_DELAY;
+  }
+
+  // current moment is a delayed version:
+  moment = moments[select_ind];
+
+  // update the moment's ind:
+  moment_ind++;
+  if(moment_ind >= MOMENT_DELAY) {
+  moment_ind = 0;
+  }
+  */
+  // moment = Ix *(-0.000553060716181365 * (stabilization_cmd[COMMAND_ROLL]-ins_flow.lp_roll_command) -3.23315441805895 * OF_X[OF_ANGLE_DOT_IND]);
+  moment = 0;
+#endif
+
+  // printf("Predicted moment = %f, gyro = %f\n", moment, dt * (ins_flow.lp_gyro_roll - ins_flow.lp_gyro_bias_roll) * (M_PI/180.0f) / 74.0f);
+
+
+  // propagate the state with Euler integration:
+  DEBUG_PRINT("Before prediction: ");
+  if (DEBUG_INS_FLOW) { print_ins_flow_state(); }
+  if (CONSTANT_ALT_FILTER) {
+    OF_X[OF_V_IND] += dt * (g * tan(OF_X[OF_ANGLE_IND]));
+    if (OF_DRAG) {
+      // quadratic drag acceleration:
+      drag = dt * kd * (OF_X[OF_V_IND] * OF_X[OF_V_IND]) / mass;
+      // apply it in the right direction:
+      if (OF_X[OF_V_IND] > 0) { OF_X[OF_V_IND] -= drag; }
+      else { OF_X[OF_V_IND] += drag; }
+    }
+
+    /* // if we use gyros here, the angle dot estimate is ignored:
+     * if(OF_USE_GYROS) {
+    // OF_X[OF_ANGLE_IND] += dt * (ins_flow.lp_gyro_roll - ins_flow.lp_gyro_bias_roll) * (M_PI/180.0f) / 74.0f; // Code says scaled by 12, but... that does not fit...
+    } */
+
+    // temporary insertion of gyro estimate here, for quicker effect:
+    // OF_X[OF_ANGLE_IND] += dt * -2.0e-03 * (stabilization_cmd[COMMAND_ROLL]-ins_flow.lp_roll_command);
+
+    OF_X[OF_ANGLE_IND] += dt * OF_X[OF_ANGLE_DOT_IND];
+    OF_X[OF_ANGLE_DOT_IND] += dt * (moment / Ix);
+
+    if (OF_TWO_DIM) {
+      // Second axis, decoupled formulation:
+      OF_X[OF_VX_IND] += dt * (g * tan(OF_X[OF_THETA_IND]));
+      if (OF_DRAG) {
+        // quadratic drag acceleration:
+        drag = dt * kd * (OF_X[OF_VX_IND] * OF_X[OF_VX_IND]) / mass;
+        // apply it in the right direction:
+        if (OF_X[OF_VX_IND] > 0) { OF_X[OF_VX_IND] -= drag; }
+        else { OF_X[OF_VX_IND] += drag; }
+      }
+      // TODO: here also a moment estimate?
+      // TODO: add a THETA_DOT_IND
+      OF_X[OF_THETA_IND] += dt * (ins_flow.lp_gyro_pitch - ins_flow.lp_gyro_bias_pitch) *
+                            (M_PI / 180.0f) / 74.0f; // Code says scaled by 12, but... that does not fit...
+    }
+
+    DEBUG_PRINT("Rate p = %f, gyro p = %f\n", rates->p,
+                (ins_flow.lp_gyro_roll - ins_flow.lp_gyro_bias_roll) * (M_PI / 180.0f) / 74.0f);
+  } else {
+    // make sure that the right hand state terms appear before they change:
+    OF_X[OF_V_IND] += dt * (thrust * sin(OF_X[OF_ANGLE_IND]) / mass);
```

Comment:
A lot of trigonometric functions are used in the file (on float), maybe replace `sin` and `cos` by `sinf` and `cosf`.